framework.py

```diff
@@ -20,5 +20,33 @@ def init_framework(self):
     
     def destroy_framework(self):
         self.log_debug("%s: Destroying..." % self)
+
+    ##########################################################################################
+    # public methods
+
+    def get_hyperlink_html(self, url, name):
```

Comment:
It feels overkill to me to have a utils module. Since we have this object that acts as the top-level parent of the framework, why not use it? It's easily accessible from both within the framework and from apps/engines that use it.